build/lib/test/test.js

```diff
@@ -341,6 +342,154 @@ async function killiOSSimulator() {
 	});
 }
 
+class DeviceTestDetails {
+	/**
+	 *
+	 * @param {string} name device/emulator name (as reported via log output. Defaults to empty string)
+	 * @param {'emulator'|'device'|'simulator'} target
+	 * @param {string} snapshotDir
+	 * @param {Promise[]} snapshotPromises
+	 */
+	constructor(name, target, snapshotDir, snapshotPromises) {
+		this.name = name;
+		this.results = [];
+		// FIXME: These things don't feel right to pass in here.
+		this.target = target;
+		this.snapshotDir = snapshotDir;
+		this.snapshotPromises = snapshotPromises;
+		// set to completed as true by default because for multi-device build we'll end up with default entry that just corresponds to
+		// log output from CLI or not from a specific device. If the device specific test suites finish we want to ignore this one
+		// We'll set to false as soon as we see a version or test start
+		this.completed = true;
+		this.resetTestState();
+	}
+
+	appendStderr(line) {
+		this.stderr += line.trim() + '\n';
+	}
+
+	/**
+	 * 	We saw test end before, but failed to parse as JSON because we got partial output, so continue
+	 * trying until it's happy (and resets sawTestEnd to false)
+	 * @param {string} token
```

Comment:
<!--
  0 failure: 
  1 warning:  build/lib/test/te...
  
  
  DangerID: danger-id-Danger;
  File: build/lib/test/test.js;
  Line: 374;
-->

- :warning: build/lib/test/test.js line 374 – Missing JSDoc parameter description for 'token'. (valid-jsdoc)